src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,117 +54,132 @@ public String getUrlForPage(String wikiName, String pageName) {
   }
 
-  public String getUrlForPath(String wikiName, String wikiPath, EnvType envType) {
-    return addPathToUrl(getUrlForWiki(wikiName, envType), wikiPath);
+  public String getUrlForPath(String wikiPath) {
+    return getUrlForPath(Configuration.getWikiName(), Configuration.getLanguage(), wikiPath);
   }
 
   protected String addPathToUrl(String url, String path) {
-
-    String outputUrl = (!path.startsWith("/")) ? String.format("%s/%s", url, path): String.format("%s%s", url, path);
+    HttpUrl.Builder urlBuilder = HttpUrl.parse(url).newBuilder();
+    urlBuilder.addEncodedPathSegments(path).build();
 
     String qs = Configuration.getQS();
     if (StringUtils.isNotBlank(qs)) {
-      outputUrl = appendQueryStringToURL(outputUrl, qs);
+      urlBuilder.encodedQuery(qs);
     }
-    return outputUrl;
+    return urlBuilder.build().toString();
   }
 
-  public String getUrlForPath(String wikiPath) {
-    return getUrlForPath(Configuration.getWikiName(), wikiPath);
+  public String getUrlForWiki() {
+    return getUrlForWiki(Configuration.getWikiName(), Configuration.getLanguage(), false);
   }
 
-  public String getUrlForWiki() {
-    return getUrlForWiki(Configuration.getWikiName(), false);
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW) {
+    return getUrlForWiki(wikiName, language, addWWW, envType);
   }
 
   public String getUrlForWiki(String wikiName) {
     return getUrlForWiki(wikiName, false);
   }
 
-  public String getUrlForWiki(String wikiName, EnvType envType) {
-    return getUrlForWiki(wikiName, false, envType);
-  }
-
-  public String getUrlForWiki(boolean addWWW) {
-    return getUrlForWiki(Configuration.getWikiName(), addWWW);
-  }
-
   public String getUrlForWiki(String wikiName, boolean addWWW) {
     return getUrlForWiki(wikiName, addWWW, envType);
   }
 
 
   public String getUrlForWiki(String wikiName, boolean addWWW, EnvType envType) {
+    return getUrlForWiki(wikiName, DEFAULT_LANGUAGE, addWWW, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language) {
+    return getUrlForWiki(wikiName, language, false, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW, EnvType envType) {
     final String wikiaName = getWikiaGlobalName(wikiName);
 
-    String www = "";
-    if (addWWW) {
-      www = "www.";
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+
+    String www = addWWW ? "www." : "";
+    String host = getFormattedWikiHost(www, wikiaName, envType, language);
+
+    if (!("en").equals(language) && forceLanguageInPath) {
+        urlBuilder.addEncodedPathSegments(language);
     }
+    return urlBuilder.scheme(getUrlProtocol()).host(host).build().toString();
+  }
+
+  private String getFormattedWikiHost(String www, String wikiaName, EnvType envType, String language) {
+    if (!forceLanguageInPath)
+      wikiaName = String.join(".", language, wikiaName);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "wikiaName". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)